academy/exchange/cloud/client.py

```diff
@@ -1,656 +1,377 @@
+        self._uid = uid
+        self._exchange = exchange
+
+        response = self.exchange._session.get(
+            self.exchange._mailbox_url,
+            json={'mailbox': uid.model_dump_json()},
+        )
+        response.raise_for_status()
+        data = response.json()
+        if not data['exists']:
+            raise BadEntityIdError(uid)
+
+    @property
+    def exchange(self) -> HttpExchange:
+        """Exchange client."""
+        return self._exchange
+
+    @property
+    def mailbox_id(self) -> EntityId:
+        """Mailbox address/identifier."""
+        return self._uid
+
+    def close(self) -> None:
+        """Close this mailbox client.
+
+        Warning:
+            This does not close the mailbox in the exchange. I.e., the exchange
+            will still accept new messages to this mailbox, but this client
+            will no longer be listening for them.
+        """
+        pass
+
+    def recv(self, timeout: float | None = None) -> Message:
+        """Receive the next message in the mailbox.
+
+        This blocks until the next message is received or the mailbox
+        is closed.
+
+        Args:
+            timeout: Optional timeout in seconds to wait for the next
+                message. If `None`, the default, block forever until the
+                next message or the mailbox is closed.
+
+        Raises:
+            MailboxClosedError: if the mailbox was closed.
+            TimeoutError: if a `timeout` was specified and exceeded.
+        """
+        try:
+            response = self.exchange._session.get(
+                self.exchange._message_url,
+                json={'mailbox': self.mailbox_id.model_dump_json()},
+                timeout=timeout,
+            )
+        except requests.exceptions.Timeout as e:
+            raise TimeoutError(
+                f'Failed to receive response in {timeout} seconds.',
+            ) from e
+        if response.status_code == _FORBIDDEN_CODE:
+            raise MailboxClosedError(self.mailbox_id)
+        response.raise_for_status()
+
+        message = BaseMessage.model_from_json(response.json().get('message'))
+        logger.debug(
+            'Received %s to %s',
+            type(response).__name__,
+            self.mailbox_id,
+        )
+        return message
+
+
+@contextlib.contextmanager
+def spawn_http_exchange(
+    host: str = '0.0.0.0',
+    port: int = 5463,
+    *,
+    level: int | str = logging.WARNING,
+    timeout: float | None = None,
+) -> Generator[HttpExchange]:
+    """Context manager that spawns an HTTP exchange in a subprocess.
+
+    This function spawns a new process (rather than forking) and wait to
+    return until a connection with the exchange has been established.
+    When exiting the context manager, `SIGINT` will be sent to the exchange
+    process. If the process does not exit within 5 seconds, it will be
+    killed.
+
+    Note: The exclusion of authentication and ssl configuration is

```

Comment:
Probably worth making this a warning block. Also, it might need to be indented to render as an admonition (not sure).
```
Warning:
    The exclusion...
```